contributing.md

```diff
@@ -0,0 +1,163 @@
+# Thank you!
+First off, thank you for considering contributing to `beets`. It's people like you that continue to allow `beets` to succeed.
+
+Following these guidelines helps to communicate that you respect the time of the developers managing and developing this open source project. In return, they should reciprocate that respect in addressing your issue, assessing changes, and helping you finalize your pull requests.
+
+# Types of contributions
+
+`beets` is an open source project and we love to receive contributions from our community — you! There are many ways to contribute, programmer or not.
+
+## Non-programming contribution ideas
+* Promote beets! Help get the word out by telling your friends, writing a blog
+  post, or discussing it on a forum you frequent.
+* Improve the [documentation][docs]. It's incredibly easy to contribute here:
+  just find a page you want to modify and hit the "Edit on GitHub" button in
+  the upper-right. You can automatically send us a pull request for your
+  changes.
+* GUI design. For the time being, beets is a command-line-only affair. But
+  that's mostly because I don't have any great ideas for what a good GUI
+  should look like. If you have those great ideas, please get in touch.
+* Benchmarks. We'd like to have a consistent way of measuring speed
+  improvements in beets' tagger and other functionality as well as a way of
+  comparing beets' performance to other tools. You can help by compiling a
+  library of freely-licensed music files (preferably with incorrect metadata)
+  for testing and measurement.
+* Think you have a nice config or unique use-case for beets? We'd love to hear about it! Submit a post to our [our forums](https://discourse.beets.io/) for a chance to get featured in our blog or use-case page.
```

Comment:
This is something that obviously isn't already implemented, but I think could be a cool idea to have a wiki page or similar that highlights unique or interesting things people do with beets

Reply:
Yeah, seems cool! There was even a proposal to do something quite similar over in #2386. Nobody actually picked up the mantle but I still think it's a very cool idea.

Reply:
Yeah, for example, [flexget ](https://flexget.com/) maintains a [cookbook](https://flexget.com/Cookbook) for useful "recipes" (configurations) for both advanced stuff and more common simple use cases.

The "meet the devs/users" thing that post suggests is another cool idea, and perhaps the next step after implementing something like this.

Reply:
In the effort of being actionable, and since I'm bad at coming up with names, I propose we add a label in discourse called 'recipe' then promote people to submit any cool or unique configurations or "recipes" to discourse that could then be added under a wiki page called "cookbook".

Reply:
As for a cookbook specifically, we do have a section of the docs that was intended to be something somewhat similar:
https://beets.readthedocs.io/en/stable/guides/advanced.html

Although posting in the Discourse is a lower-effort alternative. I like the Discourse idea, and we can "graduate" things to the docs if they need even more visibility. :smiley:

Reply:
Made a new category on discourse called [ "Show and Tell"](https://discourse.beets.io/t/about-the-show-and-tell-category/1377) and advertised it [here](https://discourse.beets.io/t/new-categories/1378).